specs/0108-community-health-dashboard.md

```diff
@@ -0,0 +1,49 @@
+- Feature Name: Community Health Dashboard
+- Created: 2020-10-23
+- Associated Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1672966
+
+# Summary
+
+Create a team-specific health dashboard targeted at Pontoon users with Manager rights.
+
+# Motivation
+
+Team managers as well as project managers should have the ability to understand how localization communities are performing. Team dashboards already provide some of the answers, including the overall status of completion and translation statistics, but the information is only available in the form of a current snapshot.
+
+There's no historic data to provide insights into how the community health is developing. Missing is the information on how completion and volume of unreviewed strings are changing over time. It's not clear what share of translations was copied from the Machinery. We also don't know the ratio of translations approved without a peer review.
+
+# Out of scope
+
+At this stage we’re only interested in presenting data we already collect in Pontoon. We're primarily focusing on exposing data to team managers; project managers will be able to obtain data across all teams if needed, e.g. by using scripts and the API.
+
+# Feature explanation
+
+A new "Insights" tab is made available on the Team page, positioned next to the "Contributors" tab. It consists of several sections, presenting data for the period of the last 12 months. Each section is described below. Fore more details about the design, see the [Mockup](#mockup) section.
+
+## Active users
+
+Pie charts showing ratios of active vs. all managers, reviewers and contributors. See the existing scripts ([1](https://github.com/flodolo/scripts/blob/954fa85/pontoon/active_contributors.py), [2](https://github.com/flodolo/scripts/blob/954fa85/pontoon/list_reviewers_with_contribution_stats.py)) for criteria of defining active users and for details on how to retrieve data.
```

Comment:
We should expect some confusion around introducing the term "reviewers" as a role, when that role doesn't existing within Pontoon. We could either formalize that change by renaming the "Translator" role to "Reviewer" or lean more heavily into the ambiguity and change "reviewer" to "translator" for consistency.

Reply:
I personally think the latter would be more confusing, because the review is done by both manager and translators. Maybe "Active reviewers", to imply that it's an action and not a role?

Reply:
That could work, with a tooltip that explains how active reviewers are defined?

Reply:
Yup, that is confusing. I don't know if adding "Active" would solve the problem, because the whole section is already called "Active (users)" and the charts imply "Active" managers/contributors/reviewers using the "(out of N)" block.

If we're splitting users by activity type, Contributors is also not a great name - it implies any kind of activity, but it only includes users that submitted translations or suggestions. What about Authors? That should also reduce the confusion with user roles.

And I just realized the spec should explicitly define what is an active manager, because the scripts we link to don't cover that category. I assume active manager is a manager who authored or reviewed a translation in a period of time?

I'm also in favor of adding a [tooltip or some other form of description](https://github.com/mozilla/pontoon/pull/1716#discussion_r512501074) with more details.

Reply:
I've thought we needed to revisit the naming for roles for a while to make them more descriptive of our expectation of their principle function (e.g., Contributor -> Translator, Translator -> Reviewer). If this is the opportunity to talk about it, let's go for it. Probably in a different forum though.

We previously had defined an active manager as a manager who logged into Pontoon within a certain time frame.

Reply:
> I've thought we needed to revisit the naming for roles for a while to make them more descriptive of our expectation of their principle function (e.g., Contributor -> Translator, Translator -> Reviewer). If this is the opportunity to talk about it, let's go for it. Probably in a different forum though.

I'm open for that. Are you also suggesting we should split these charts by role instead of activity type?

> We previously had defined an active manager as a manager who logged into Pontoon within a certain time frame.

I saw that in the scoping doc. Isn't that somewhat low barrier to become active compared to other two categories?

Reply:
I more mean that the role names should reflect their contribution/activity. The way the chart is split is fine, it's the larger issue of role names. Not sure what Flod thinks?

Reply:
I'm basing naming on role expectations. We don't expect a manager's primary contribution to be a translation, it's to keep their community healthy and overcome bottlenecks. We expect "Translators" to be the peer reviewers. We expect "contributors" to suggest translations.

Reply:
I added a note about detailed descriptions for each data series.

To avoid blocking the OKR, let's save renaming of roles (and possibly other related terms) for another occasion.